src/main/java/com/vaadin/training/bugrap/ui/MainView.java

```diff
@@ -31,19 +41,60 @@ public void enter(final ViewChangeEvent event) {
 		initProjectSelect();
 		initSearchReportsDropdown();
 		initProjectVersions();
+		initReportsGrid();
 
+		subscribeToEvents();
 	}
 
-	private void initProjectVersions() {
-		findProjectVersionsAndSelectEmpty();
-		UIEventBus.getInstance().subscribe(ProjectChangedEvent.class, this::receive);
+	private void subscribeToEvents() {
+		final EventBus eventBus = UIEventBus.getInstance();
+		eventBus.subscribe(ProjectChangedEvent.class, this::receiveProjectChangedEvent);
+		eventBus.subscribe(ProjectVersionChangedEvent.class, this::receiveVersionChangedEvent);
 	}
 
-	private void findProjectVersionsAndSelectEmpty() {
+	private void initReportsGrid() {
+
+		reportsGridLayout.removeComponent(reportsGrid);
```

Comment:
You can also keep the report in the layout all the time, but in case of report list being empty just set reportsGrid.setVisible(false).
This would enable you to have the whole structure of the view in you design-layout.